Sources/HeliumLogger/HeliumLogger.swift

```diff
@@ -61,16 +61,133 @@ public class HeliumLogger {
     /// If true, use the detailed format when a user logging format wasn't specified.
     public var details: Bool = true
 
+    /// If true, use the full file path, not just the filename.
+    public var fullFilePath: Bool = false
+
     /// If not nil, specifies the user specified logging format.
-    public var format: String?
-    
+    /// For example: "[(%date)] [(%type)] [(%file):(%line) (%func)] (%msg)"
+    public var format: String? {
+        didSet {
+            if let format = self.format {
+                customFormatter = HeliumLogger.parseFormat(format)
+            } else {
+                customFormatter = nil
+            }
+        }
+    }
+
     /// If not nil, specifies the format used when adding the date and the time to the
     /// logged messages
-    public var dateFormat: String?
+    public var dateFormat: String? {
+        didSet {
+            dateFormatter = HeliumLogger.getDateFormatter(format: dateFormat, timeZone: timeZone)
+        }
+    }
+
+    /// If not nil, specifies the timezone used in the date time format
+    public var timeZone: TimeZone? {
```

Comment:
If we are adding support for logging to custom timezones, I think we need to include the timezone in the `defaultDateFormat`. Otherwise, it won't be clear in the logs what the timezone is.